ext/nnstreamer/tensor_decoder/tensordec-flexbuf.cc

```diff
@@ -0,0 +1,161 @@
+ *
+ * @see         https://github.com/nnstreamer/nnstreamer
+ * @author      Gichan Jang <gichan2.jang@samsung.com>
+ * @bug         No known bugs except for NYI items
+ *
+ */
+
+#include <glib.h>
+#include <nnstreamer_log.h>
+#include <nnstreamer_plugin_api.h>
+#include <nnstreamer_plugin_api_decoder.h>
+#include "tensordecutil.h"
+#include <flatbuffers/flexbuffers.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+void init_flxb (void) __attribute__ ((constructor));
+void fini_flxb (void) __attribute__ ((destructor));
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static int
+flxb_init (void **pdata)
+{
+  *pdata = NULL;
+  return TRUE;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static void
+flxb_exit (void **pdata)
+{
+  return;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static int
+flxb_setOption (void **pdata, int opNum, const char *param)
+{
+  return TRUE;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static GstCaps *
+flxb_getOutCaps (void **pdata, const GstTensorsConfig *config)
+{
+  GstCaps *caps;
+  caps = gst_caps_from_string (GST_FLEXBUF_CAP_DEFAULT);
+  setFramerateFromConfig (caps, config);
+  return caps;
+}
+
+/** @brief tensordec-plugin's GstTensorDecoderDef callback */
+static GstFlowReturn
+flxb_decode (void **pdata, const GstTensorsConfig *config,
+    const GstTensorMemory *input, GstBuffer *outbuf)
+{
+  GstMapInfo out_info;
+  GstMemory *out_mem;
+  unsigned int i, num_tensors = config->info.num_tensors;
+  size_t flex_size;
+  flexbuffers::Builder fbb;
+
+  fbb.Map ([&]() {
+    fbb.UInt ("num_tensors", num_tensors);
+    fbb.Int ("rate_n", config->rate_n);
+    fbb.Int ("rate_d", config->rate_d);
+    for (i = 0; i < num_tensors; i++) {
+      gchar * tensor_key = g_strdup_printf ("tensor_%d", i);
+      gchar * tensor_name = NULL;
+
+      if (config->info.info[i].name == NULL) {
+        tensor_name = g_strdup ("");
+      } else {
+        tensor_name = g_strdup (config->info.info[i].name);
+      }
+      tensor_type type = config->info.info[i].type;
+
+      fbb.Vector (tensor_key, [&] () {
+        fbb += tensor_name;
+        fbb += type;
+        fbb.Vector (config->info.info[i].dimension, NNS_TENSOR_RANK_LIMIT);
+        fbb.Blob (input[i].data, (size_t) input[i].size);
+      });
+      g_free (tensor_key);
+      g_free (tensor_name);
+    }
+  });
+  fbb.Finish ();
+  flex_size = fbb.GetSize ();
+
+  g_assert (outbuf);
+
+  if (gst_buffer_get_size (outbuf) == 0) {
+    out_mem = gst_allocator_alloc (NULL, flex_size, NULL);
+  } else {
+    if (gst_buffer_get_size (outbuf) < flex_size) {
+      gst_buffer_set_size (outbuf, flex_size);
+    }
+    out_mem = gst_buffer_get_all_memory (outbuf);
+  }
+
+  if (FALSE == gst_memory_map (out_mem, &out_info, GST_MAP_WRITE)) {
+    gst_allocator_free (NULL, out_mem);
```

Comment:
plz check condition for freeing memory (line 107)
```
gboolean need_alloc = (gst_buffer_get_size (outbuf) == 0)

if (need_alloc)
out_mem = gst_allocator_alloc (NULL, flex_size, NULL);
....
```

Reply:
Thanks. Updated.